chain/client/src/client_actor.rs

```diff
@@ -1331,11 +1331,14 @@ impl ClientActor {
                     };
 
                 let me = self.client.validator_signer.as_ref().map(|x| x.validator_id().clone());
+                let block_header =
+                    unwrap_or_run_later!(self.client.chain.get_block_header(&sync_hash));
+                let prev_hash = block_header.prev_hash();
                 let shards_to_sync = (0..self.client.runtime_adapter.num_shards())
                     .filter(|x| {
                         self.client.shards_mgr.cares_about_shard_this_or_next_epoch(
                             me.as_ref(),
-                            &sync_hash,
+                            prev_hash,
```

Comment:
Why should we use prev hash here?

Reply:
hmm...not sure how this code worked before....

Reply:
I think this will only cause a bug if epochs only have one block. Maybe the state sync tests never tested that case